### PR TITLE
[ORCA] Add xshards example

### PR DESCRIPTION
**Pull request by yexinyinancy**
### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: python/orca/dev/test/run-tutorial-xshards.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ ray stop -f
 cd "`dirname $0`"
 cd ../../tutorial/xshards
 
-echo "Running Xshards tests"
+echo "Running Xshards tests 1"
```
Comment (hkvision) [inline python/orca/dev/test/run-tutorial-xshards.sh:29]: change the name to the example

File: python/orca/tutorial/xshards/titanic.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    data.loc[data['Sex'] == 0, 'SexByPclass'] = data.loc[data['Sex'] == 0, 'Pclass']
+    data.loc[data['Sex'] == 1, 'SexByPclass'] = data.loc[data['Sex'] == 1, 'Pclass'] + 3
+    return data
+data_shard = data_shard.transform_shard(change_val)
+
+# save
+data_shard.save_pickle('./result')
+
+stop_orca_context()
+
```
Comment (hkvision) [inline python/orca/tutorial/xshards/titanic.py:60]: remove extra blank line

File: python/orca/dev/test/run-tutorial-xshards.sh [hunk 3]
```diff
@@ -43,4 +43,25 @@ python tabular_playground_series.py
 now=$(date "+%s")
 time1=$((now - start))
 
-echo "Running Xshards tests time used: $time1 seconds"
+echo "Running Xshards tests 1 time used: $time1 seconds"
```
Comment (hkvision) [inline python/orca/dev/test/run-tutorial-xshards.sh:46]: put timing at the end

File: python/orca/tutorial/xshards/titanic.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+from bigdl.orca import init_orca_context, stop_orca_context
+import bigdl.orca.data.pandas
+
+init_orca_context(cluster_mode="local", cores=4, memory="3g")
+
+file_path = "titanic.csv"
+data_shard = bigdl.orca.data.pandas.read_csv(file_path)
+
+# drop
+def drop_func(df):
```
Comment (hkvision) [inline python/orca/tutorial/xshards/titanic.py:28]: drop_passenger

File: python/orca/tutorial/xshards/titanic.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    df['Cabin'] = df['Cabin'].replace(['B', 'C'], 'H')
+    df['Cabin'] = df['Cabin'].replace(['F', 'G'], 'L')
+    df['Cabin'] = df['Cabin'].map({'X': 0, 'L': 1, 'M': 2, 'H': 3})
+    df['Cabin'] = df['Cabin'].astype(int)
+    return df
+data_shard = data_shard.transform_shard(process_cabin)
+
+
+# astype, loc
+def change_val(data):
```
Comment (hkvision) [inline python/orca/tutorial/xshards/titanic.py:48]: encode

**Review by hkvision — APPROVED** [review]

**Comment by yexinyinancy** [comment]
For the source [notebook](https://www.kaggle.com/code/chuanguy/titanic-data-processing-with-python-0-813/notebook), I only select part of the code which processes the data and xshards can support now at the meantime. The remaining is some codes for plot and statistics.

**Pull request merged**